dumbbells/dataset.py

```diff
@@ -0,0 +1,110 @@
+import gym
+import torch
+from torch.utils.data import Dataset as Ds, DataLoader
+
+
+class Dataset(Ds):
+    """Dataset containing information about the environment"""
+
+    def __init__(self, game, memory_size):
+        """Constructor, with the environment and memory_size specified
+
+        Args:
+            game: String containing the name of the environment to be created. Must be a valid environment. A list of
+                  valid environments can be found at "https://gym.openai.com/envs/#classic_control", or with
+                  gym.envs.registry.all()
+            memory_size: Max images we want to store at any time.
+
+        Returns: None
+        """
+
+        self.env = gym.make(game)
+        self.memory_size = memory_size
+        self.memory = []
+        self.position = 0
+
+        # For the environment "MountainCar-v0", there are 3 actions available:
+        # 0: Accelerate to the left
+        # 1: Don't accelerate
+        # 2: Accelerate to the right
+        self.action_space = self.env.action_space
+
+        # For the environment "MountainCar-v0", there are 2 observations, presented in the following order:
+        # Car Position: Ranges from -1.2 to 0.6
+        # Car Velocity: Ranges from -0.07 to 0.07
+        # The car starts at some position between -0.6 to -0.4
+        self.state_space = self.env.reset()
+
+        # For the environment "MountainCar-v0", there are 2 possible rewards:
+        # Reward of 0 is awarded if the agent has reached the flag (position 0.5)
+        # Reward of -1 is awarded if the agent is at a position < 0.5
+        self.reward_space = [-1, 0]
```

Comment:
Is the reward space the same for all environments, or could we somehow get it out of the `env` itself?

Reply:
I don't think that gym allows for extracting the reward. From what I saw in the documentation and the GitHub for mountain_car.py, it was just a float that is returned, either -1 or 0. There isn't really a variable we can pull from that I could see.